physical/etcd.go

```diff
@@ -79,10 +79,23 @@ func newEtcdBackend(conf map[string]string) (Backend, error) {
 	if address, ok := conf["address"]; ok {
 		machines = address
 	}
+	machinesParsed := strings.Split(machines, EtcdMachineDelimiter)
 
-	// Create a new client from the supplied addres and attempt to sync with the
+	// Create a new client from the supplied address and attempt to sync with the
 	// cluster.
-	client := etcd.NewClient(strings.Split(machines, EtcdMachineDelimiter))
+	var client *etcd.Client
+	cert, has_cert := conf["tls_cert_file"]
+	key, has_key := conf["tls_key_file"]
+	ca, has_ca := conf["tls_ca_file"]
+	if has_cert || has_key || has_ca {
```

Comment:
This probably needs to check for `(has_cert && has_key) || has_ca`; if you have a client cert but no key, or vice versa, you can't actually auth.